### PR TITLE
Replace Pipe type with Callable in Language

### PR DESCRIPTION
**Pull request by adrianeboyd**
## Description

Replace `Pipe` type with `Callable[[Doc], Doc]` in `Language`.

Requires mypy 0.990 with support for `hasattr`.

### Types of change

Types.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by polm** [comment]
Technically a Pipe has a few more features than a Callable, like having a factory associated with it, right? I tried to see if that would cause any issues in Language, but I couldn't come up with any examples. 

Separately from that, docstrings still refer to `Pipe`. Do we want to keep them that way?

File: spacy/language.py [hunk 1]
```diff
@@ -1363,7 +1362,7 @@ def resume_training(self, *, sgd: Optional[Optimizer] = None) -> Optimizer:
 
     def set_error_handler(
         self,
-        error_handler: Callable[[str, "Pipe", List[Doc], Exception], NoReturn],
+        error_handler: Callable[[str, PipeCallable, List[Doc], Exception], NoReturn],
```
Comment (polm) [inline spacy/language.py:1365]: The docstring here is an example of one where the function signature in the docstring and the actual one no longer match. Do we want to update this?

**Comment by adrianeboyd** [comment]
I intentionally left the docstrings alone initially.

I don't think the docstrings should have `PipeCallable`. I think they could have `Callable` or `Callable[[Doc], Doc]`. Or we can just replace it all with `Callable[[Doc], Doc]` maybe and drop `PipeCallable`?

I don't think they should have `Pipe`, which isn't accurate.

**Comment by adrianeboyd** [comment]
Using `Callable[[Doc], Doc]`, which also matches the API docs.

**Review by polm — APPROVED** [review]
Looks good.

**Pull request merged**